FirebaseAdmin/FirebaseAdmin/Messaging/FirebaseMessagingClient.cs

```diff
@@ -102,11 +112,97 @@ internal sealed class FirebaseMessagingClient : IDisposable
             }
         }
 
+        /// <summary>
+        /// Sends all messages in a single batch.
+        /// </summary>
+        /// <param name="messages">The messages to be sent. Must not be null.</param>
+        /// <param name="dryRun">A boolean indicating whether to perform a dry run (validation
+        /// only) of the send. If set to true, the messages will be sent to the FCM backend service,
+        /// but it will not be delivered to any actual recipients.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
+        /// operation.</param>
+        /// <returns>A task that completes with a <see cref="SendResponse"/>, giving details about
+        /// the batch operation.</returns>
+        public async Task<SendResponse> SendAllAsync(
```

Comment:
The returned object should be called `BatchResponse` to be consistent with Java and Node.js. Sorry if I wasn't clear in my previous review. The public API should be:

```
class BatchResponse
{
  public int SuccessCount;
  public int FailureCount;
  public IReadOnlyList<SendResponse> Responses;
}

class SendResponse
{
  public string MessgeId;
  public FirebaseException Exception;
  public boolean IsSuccess;
}
```